dropwizard-testing/pom.xml

```diff
@@ -172,6 +172,10 @@
             <groupId>org.glassfish.jersey.containers</groupId>
             <artifactId>jersey-container-servlet-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.connectors</groupId>
+            <artifactId>jersey-grizzly-connector</artifactId>
+        </dependency>
```

Comment:
@zUniQueX Could you please add a remark in the upgrade notes for Dropwizard 2.1.x that `dropwizard-testing` (and more specifically `DropwizardAppExtension`) is now using the Grizzly connector instead of the built-in JDK connector?

They are unfortunately not equivalent. 😓

This being said, do you think it could make sense to use the Apache HttpClient connector in the first place? `dropwizard-client` is building on Apache HttpClient anyway.

- https://github.com/dropwizard/dropwizard/blob/v2.1.0/dropwizard-client/pom.xml#L84-L91
- https://eclipse-ee4j.github.io/jersey.github.io/documentation/2.36/user-guide.html#d0e4978
- https://mvnrepository.com/artifact/org.glassfish.jersey.connectors/jersey-apache-connector/2.36

In Dropwizard 3.x and 4.x, we could probably use the Apache HttpClient 5 connector (and migrate `dropwizard-client` to Apache HttpClient 5).

https://mvnrepository.com/artifact/org.glassfish.jersey.connectors/jersey-apache5-connector

Reply:
@joschi Thanks for the review! This PR was meant more like a consideration than a final solution. I'm not happy introducing a connector in a patch release but I also don't want Java 17 users to be blocked by this.

As I reported in the linked issue, I checked different connectors against our current codebase. The Apache connector broke some tests just in the `dropwizard-testing` module while the Grizzly connector required only one small adjustment in the tests. I thought users would have it easier then to use the Grizzly connector.

If we really want to introduce a connector in the 2.x baseline, I'll revisit this and check, what else the Apache connector will break. We could also think of bringing a version 2.2.0 quickly with some minor adjustments including this change before we release 3.x.

I'm not sure, if we would want to use the `apache5-connector` directly. This wouldn't introduce additional migration cost for 3.x. Additionally, Jersey releases can take quite some time, so we'll probably have to override the hc versions with our current ones to get `dependencyConvergence` working.